src/psl.cpp

```diff
@@ -0,0 +1,180 @@
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+#include "psl.h"
+#include "punycode.h"
+
+namespace Url
+{
+
+    PSL::PSL(std::istream& stream)
+    {
+        std::string line;
+        while (std::getline(stream, line))
+        {
+            // Skip blank lines
+            if (!line.size())
+            {
+                continue;
+            }
+
+            // Skip comments
+            if (line.size() >= 2 && line[0] == '/' && line[1] == '/')
```

Comment:
Easier to understand as `if (line.compare(0, 2, "//") == 0)`
But what about comment lines that don't begin as the first character?
`if (line.compare(line.find_first_not_of(" \t"), 2, "//") == 0)`